purchase_isolated_quotation/models/purchase.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# © 2017 Ecosoft (ecosoft.co.th).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    is_order = fields.Boolean(  # False is quotation, True is PO.
+        string='Order type',
+        readonly=True,
+        index=True,
+        default=lambda self: self._context.get('is_order', False),
+    )
+    quote_id = fields.Many2one(
+        'purchase.order',
+        string='Quotation Reference',
+        readonly=True,
+        ondelete='restrict',
+        copy=False,
+    )
+    order_id = fields.Many2one(
+        'purchase.order',
+        string='Order Reference',
+        readonly=True,
+        ondelete='restrict',
+        copy=False,
+    )
+    state2 = fields.Selection(
```

Comment:
better to name it quotation_state ?

Reply:
i agree with you, it will be meaningful name.